build.sbt

```diff
@@ -271,14 +271,22 @@ def macroCompileSettings: Seq[Setting[_]] = Seq(
   }
 )
 
-def mimaSettings(versions: Seq[String]): Seq[Setting[_]] = {
+val version140 = "1.4.0"
+val version1412 = "1.4.12"
+
+def mimaSettings(since: String): Seq[Setting[_]] = {
+  val versions = Seq(since)
   Seq(
     mimaPreviousArtifacts := {
       scalaVersionFilter(scalaBinaryVersion.value, versions).map { version =>
         organization.value % s"${moduleName.value}_${scalaBinaryVersion.value}" % version
       }.toSet
     },
-    mimaBinaryIssueFilters += ProblemFilters.exclude[Problem]("com.lightbend.lagom.internal.*")
+    mimaBinaryIssueFilters ++= Seq(
+      ProblemFilters.exclude[Problem]("com.lightbend.lagom.internal.*"),
+      ProblemFilters.exclude[Problem]("com.lightbend.lagom.*Components*"),
+      ProblemFilters.exclude[Problem]("com.lightbend.lagom.*Module*")
```

Comment:
```
      ProblemFilters.exclude[Problem]("com.lightbend.lagom.*Components*"),
      ProblemFilters.exclude[Problem]("com.lightbend.lagom.*Module*")
```

Hmm, I don't see how this works. Why is it OK to add these two filters?

Reply:
Just noticed the filter are there on `1.5.x` too (https://github.com/lagom/lagom/pull/1895/files#diff-0). Still not sure why.

Reply:
I think I copy-and-pasted it from 1.5.x. I can remove but maybe better to understand why we have it in 1.5.x and master. If it's good to have it there, it should be good to have it here. :-)

Reply:
The `*Module*` exclusion, is included in #1385, which is actually a rework of #1360. I think it makes sense because we can consider guice modules to be internal API.

The `*Components*` exclusion was added on #1483. There is a comment from Tim saying:

> I thought we decided to filter all of the "Components" classes out of the MiMa check (since third-party libraries shouldn't depend on them directly

I consider both to be valid exclusions.